### PR TITLE
[Doris On ES][Bug-Fix] split es publish_address if it has host

### LINKED ISSUE #5949: [Doris On ES][Bug] publish_address has hostname, it causes  bug
**Describe the bug**
https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/http/HttpInfo.java
当es publish_address包含host，则不能连接到es集群。EsNodeInfo需要剔除host

### PR DESCRIPTION
**Pull request by ChPi**
## Proposed changes
#5949 

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by stalary** [comment]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/EsNodeInfo.java [hunk 1]
```diff
@@ -70,6 +70,9 @@ public EsNodeInfo(String id, Map<String, Object> map, boolean httpSslEnabled) {
         if (httpMap != null) {
             String address = (String) httpMap.get("publish_address");
             if (address != null) {
+                if (address.contains("/")) {
```
Comment (wuyunfeng) [inline fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/EsNodeInfo.java:73]: maybe we can split this address by `/` directly,

**Review by wuyunfeng — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/EsNodeInfo.java (hunk 1 above)
Comment (ChPi) [reply fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/EsNodeInfo.java]: yes, we don't need to use contains, use one-liner code.  i modified it.

File: fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/EsNodeInfo.java [hunk 2]
```diff
@@ -70,6 +70,7 @@ public EsNodeInfo(String id, Map<String, Object> map, boolean httpSslEnabled) {
         if (httpMap != null) {
             String address = (String) httpMap.get("publish_address");
             if (address != null) {
+                address = address.substring(address.lastIndexOf('/') + 1);
```
Comment (wuyunfeng) [inline fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/EsNodeInfo.java:73]: add comments for this line, give an simple conf-example

**Review by wuyunfeng — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/EsNodeInfo.java (hunk 2 above)
Comment (wuyunfeng) [reply fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/EsNodeInfo.java:73]: Provide UT-test for this case?

**Review by wuyunfeng — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/EsNodeInfo.java (hunk 2 above)
Comment (ChPi) [reply fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/EsNodeInfo.java:75]: added comments and unit tests for my fix. Please review. Thanks.

File: fe/fe-core/src/test/java/org/apache/doris/external/elasticsearch/EsNodeInfoTest.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
+/*********************************************************************************
```
Comment (wuyunfeng) [inline fe/fe-core/src/test/java/org/apache/doris/external/elasticsearch/EsNodeInfoTest.java:1]: reference  other source file and replace line 1-7 with Apache license,

**Review by wuyunfeng — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/test/java/org/apache/doris/external/elasticsearch/EsNodeInfoTest.java (hunk 3 above)
Comment (ChPi) [reply fe/fe-core/src/test/java/org/apache/doris/external/elasticsearch/EsNodeInfoTest.java]: 啊自动生成公司Copyright没有修改，合并的时候这条PR commit历史会消失吗

**Review by wuyunfeng — APPROVED** [review]

File: fe/fe-core/src/test/java/org/apache/doris/external/elasticsearch/EsNodeInfoTest.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+
+public class EsNodeInfoTest  extends EsTestCase{
+
+    @Test
+    public void parsePublishAddressTest() throws Exception {
+        ObjectMapper mapper = new ObjectMapper();
+        JsonParser jsonParser = mapper.getJsonFactory().createJsonParser(loadJsonFromFile("data/es/test_nodes_http.json"));
```
Comment (wuyunfeng) [inline fe/fe-core/src/test/java/org/apache/doris/external/elasticsearch/EsNodeInfoTest.java:34]: 总感觉这个地方有点别扭

Comment (ChPi) [reply fe/fe-core/src/test/java/org/apache/doris/external/elasticsearch/EsNodeInfoTest.java:34]: 读取json吗。这我是看PartitionPhaseTest写的，json文件添加了个b节点来测试publish_address

Comment (wuyunfeng) [reply fe/fe-core/src/test/java/org/apache/doris/external/elasticsearch/EsNodeInfoTest.java:34]: 是这个JsonParser

**Pull request merged**